variables.tf

```diff
@@ -80,32 +80,14 @@ variable "service_account_create" {
 
 variable "service_account_name" {
   type        = string
-  default     = "<$addon-name>"
-  description = "The k8s <$addon-name> service account name"
+  default     = "kyverno"
+  description = "The k8s kyverno service account name"
 }
 
 variable "irsa_role_create" {
-  type        = bool
-  default     = true
-  description = "Whether to create IRSA role and annotate service account"
-}
-
-variable "irsa_policy_enabled" {
-  type        = bool
-  default     = true
-  description = "Whether to create opinionated policy to allow operations on specified zones in `policy_allowed_zone_ids`."
-}
-
-variable "irsa_assume_role_enabled" {
   type        = bool
   default     = false
-  description = "Whether IRSA is allowed to assume role defined by irsa_assume_role_arn."
-}
-
-variable "irsa_assume_role_arn" {
-  type        = string
-  default     = ""
-  description = "Assume role arn. Assume role must be enabled."
+  description = "Whether to create IRSA role and annotate service account"
 }
```

Comment:
We have `var.irsa_role_crete` set to `true` by default in [addon template](https://github.com/lablabs/terraform-aws-eks-addon-template/blob/207e462feae640aa688f72b0770dbc5bd8d6000d/variables.tf#LL87-L91C2)

Is it OK to break the convention?

Reply:
Neither Kyverno nor [Policy Reporter](https://github.com/lablabs/terraform-aws-eks-kyverno-policy-reporter/pull/1) require IAM roles at the moment. It seems unnecessary to create them by default. 
@dojci What do you think?

Reply:
@tomas-balaz good point. This setting has a larger context regarding security implications. We should restrict by default access to the instance profile assigned to the worker node for pods that do not need these permissions. There are multiple ways how we can achieve this in cluster:
1. Use dedicated service accounts and IAM role for each application to prevent using instance profile role
2. [Block access to instance metadata](https://aws.github.io/aws-eks-best-practices/security/docs/iam/#restrict-access-to-the-instance-profile-assigned-to-the-worker-node) by requiring the instance to use IMDSv2 only and updating the hop count to 1, but setting can be modified only in the node group's launch template.
3. We can also block a pod's access to EC2 metadata by using NetworkPolicies to prevent access from all pods and allow access only to those that need access to the EC2 metadata service. This method is challenging to apply in the context of a whole cluster with vanilla tooling.

So as the simplest and most universal solution to prevent pods from inheriting instance profile IAM roles, we should keep the creation of default IRSA role and service account enabled by default.

Reply:
Updated @dojci @tomas-balaz
Pls review both this and Policy Reporter